ios/BrightcovePlayer.m

```diff
@@ -129,20 +143,110 @@ - (void)playbackController:(id<BCOVPlaybackController>)controller playbackSessio
+        }
+    } else if (lifecycleEvent.eventType == kBCOVPlaybackSessionLifecycleEventFailedToPlayToEndTime) {
+        if (self.onStatusEvent) {
+            self.onStatusEvent(@{
+                                 @"type": @("failedToPlayToEndTime"),
+                                 @"error": lifecycleEvent.properties[kBCOVPlaybackSessionLifecycleEventError]
+                                 });
+        }
+    }  else if (lifecycleEvent.eventType == kBCOVPlaybackSessionLifecycleEventResumeBegin) {
+        if (self.onStatusEvent) {
+            self.onStatusEvent(@{
+                                 @"type": @("resumeAttemptBegin")
+                                 });
+        }
+    }  else if (lifecycleEvent.eventType == kBCOVPlaybackSessionLifecycleEventResumeComplete) {
+        if (self.onStatusEvent) {
+            self.onStatusEvent(@{
+                                 @"type": @("resumeAttemptComplete")
+                                 });
+        }
+    } else if (lifecycleEvent.eventType == kBCOVPlaybackSessionLifecycleEventResumeFail) {
+        if (self.onStatusEvent) {
+            self.onStatusEvent(@{
+                                 @"type": @("resumeAttemptFail")
+                                 });
+        }
+    } else if (lifecycleEvent.eventType == kBCOVPlaybackSessionLifecycleEventPlaybackStalled) {
+        // name non-standard to avoid conflict with RCTVideo event PlaybackStalled
+        if (self.onStatusEvent) {
+            self.onStatusEvent(@{
+                                 @"type": @("playbackStalled")
+                                 });
+        }
+    } else if (lifecycleEvent.eventType == kBCOVPlaybackSessionLifecycleEventPlaybackRecovered) {
+        if (self.onStatusEvent) {
+            self.onStatusEvent(@{
+                                 @"type": @("playbackRecovered")
+                                 });
+        }
+    } else if (lifecycleEvent.eventType == kBCOVPlaybackSessionLifecycleEventPlaybackBufferEmpty) {
+        if (self.onStatusEvent) {
+            self.onStatusEvent(@{
+                                 @"type": @("playbackBufferEmpty")
+                                 });
+        }
+    }  else if (lifecycleEvent.eventType == kBCOVPlaybackSessionLifecycleEventPlaybackLikelyToKeepUp) {
+        if (self.onStatusEvent) {
+            self.onStatusEvent(@{
+                                 @"type": @("playbackBufferLikelyToKeepUp")
+                                 });
+        }
+    } else if (lifecycleEvent.eventType == kBCOVPlaybackSessionLifecycleEventTerminate) {
+        if (self.onStatusEvent) {
+            self.onStatusEvent(@{
+                                 @"type": @("playerTerminated")
+                                 });
+        }
+    } else if (lifecycleEvent.eventType == kBCOVPlaybackSessionLifecycleEventError) {
+        if (self.onStatusEvent) {
+            self.onStatusEvent(@{
+                                 @"type": @("playerTerminated"),
```

Comment:
This should not be terminated, but "lifecycleError" or something similar. Just keeping the names consistent with the native API makes it easier if they change the name, or come up with another name that accidentally comes to be the same.